File: Data_Structure/Queue/Implement_Queue_Using_ArrayP.c
#include<stdio.h>
#include<stdlib.h>

struct queue
{
    int size;
    int front;
    int rear;
    int *arr;
};

int isFull(struct queue *q)
{
    if(q -> rear == q -> size-1)
    {
        return 1;
    }
    return 0;
}

int isEmpty(struct queue *q)
{
    if(q -> rear == q -> front)
    {
        return 1;
    }
    return 0;
}

void enQueue(struct queue *q, int val)
{
    if(isFull(q))
    {
        printf("This Queue is full");
    }
    else
    {
        q -> rear++;
        q -> arr[q -> rear] = val;
    }
    return q;
}

void deQueue(struct queue *q)
{
    int a = -1;
    if(isEmpty(q))
    {
        printf("This Queue is empty");
    }
    else
    {
        q -> front+=1;
        a = q -> arr[q -> front];     
    }
    printf("\n\nDeleted element is %d",a);
    return q;
}

void peek(struct queue *q)
{
    printf("Front element of queue is %d", q -> arr[q -> front+1]);
}

void display(struct queue *q)
{
    if(isEmpty(q))
    {
        printf("\nNo element in queue");
    }
    for(int i= q -> front; i< q->rear; i++)
        {
            printf("%d ",q -> arr[i+1]);
        }
        printf("\n\n");
}

int main()
{
    struct queue q;
    q.size = 5;
    q.front = -1;
    q.rear = -1;
    q.arr = (int *)malloc(q.size*sizeof(int));
    
    enQueue(&q, 26);
    enQueue(&q, 35);
    enQueue(&q, 56);
    enQueue(&q, 62);
    
    printf("Queue of elements is:\n");
    display(&q);
    
    peek(&q);
    deQueue(&q);
    
    printf("\n\nQueue of elements is:\n");
    display(&q);
    return 0;
}

File: Data_Structure/DS_Using_Python/Implement_Queue_Using_ArrayP.py
class Queue:
    def __init__(self, capacity):
        self.queue = [None]*capacity
        self.front = -1
        self.rear = -1
        self.capacity = capacity
        
     #Queue is full when size becomes equal to the capacity
    def isFull(self):
    	if self.rear == self.capacity-1:
    		return 1
    	return 0
    	
    #Queue is empty when size is 0 
    def isEmpty(self):
    	if self.front == self.rear:
    		return 1
    	return 0

    # Add an element
    def enqueue(self, item):
        if self.isFull():
        	print('Stack Overflow ! Cannot added new element\n')
        	return
        else:
        	self.rear += 1
        	self.queue[self.rear] = item

    # Remove an element
    def dequeue(self):
        if self.queue[0] is None:
           return None
        return self.queue.pop(0)

    # Display  the queue
    def display(self):
        if self.isEmpty():
        	print("Stack is empty !")
        else:
        	print(self.queue)

	#Size of queue
    def size(self):
        return len(self.queue)
        
    #find an element in queue
    def search(self, val):
    	for i in range(len(self.queue)):
    		   if self.queue[i] == val:
    		      print("\n\nElement found at {} position.".format(i))
    		      return
    	print("\n\nNo element found")
    	
    #top most element in queue
    def peek(self):
    	if self.queue :
    		print("\nPeek of queue is:", self.queue[0])

q = Queue(5)
q.enqueue(51)
q.enqueue(27)
q.enqueue(11)
q.enqueue(63)
q.enqueue(31)
q.enqueue(51)

print("Queue of elements is:")
q.display()

res = q.dequeue()
print("\nAfter removing an element:")
if res == None:
	 print('Element cannot be deleted')

q.display()

q.search(63)
q.peek()

sizeOfQueue = q.size()
print('\nSize of queue is', sizeOfQueue)